anchore_engine/analyzers/manager.py

```diff
@@ -29,10 +30,24 @@ def apply_hints(analyzer_report, unpackdir):
     # the config does not explicitly enable hints processing, effectively disabling
     # hints processing.
     for engine_entry in utils.content_hints(unpackdir=unpackdir):
-        pkg_type = engine_entry.get("type")
-        if pkg_type and pkg_type in syft.modules_by_engine_type:
+        pkg_type = engine_entry.get("type", None)
+
+        # Just in case it's set to null explicitly in the hint
+        pkg_type = pkg_type.lower() if pkg_type else None
+        if (
+            pkg_type
+            and pkg_type in syft.modules_by_engine_type
+            and pkg_type in hints.hints_by_type
+        ):
             handler = syft.modules_by_engine_type[pkg_type]
-            handler.save_entry(analyzer_report, engine_entry)
+            try:
+                hint = hints.hints_by_type[pkg_type](engine_entry).to_dict()
```

Comment:
If we fail on fail on processing the hints, shouldn't we bail instead of walking the remaining entries?

Reply:
Other than that question above it looks good.

Reply:
It's a single hint, we don't want one hint to cause others to fail. Also, this is just a PR to sync the branches, so wouldn't be fixing any of the underlying fixes here. If you wanna discuss in more detail about the hints let me know!